### DISCUSSION TITLE
Slow 3G模式下access偶发性失效

### DISCUSSION THREAD

**Discussion by Gumplefik** [umijs/umi#9933] [Q&A]
添加路由权限后，在慢网速的情况，unaccessible未被成功赋值到routes变量中的路由上，导致整个授权验证失效
```javascript
const [route] = useAccessMarkedRoutes(mapRoutes(newRoutes));
```

**Comment by fz6m** [comment]
自己尝试在能复现的环境里 debug 一下，看看是不是权限数据还没获取到就进路由了。

如果是，能不能保证权限数据获取完（比如 `await` 忘写了），也有可能是 react 18 auto batch 批处理导致的（`setState` 是异步的），手动 `flushSync` 一下。
